examples/siro_sandbox/app_states/app_state_tutorial.py

```diff
@@ -23,37 +20,30 @@ def __init__(
         self._sandbox_service = sandbox_service
         self._gui_agent_ctrl = gui_agent_ctrl
         self._cam_transform = None
-        self._camera_helper = CameraHelper(
-            self._sandbox_service.args, self._sandbox_service.gui_input
-        )
+        self._camera_helper = None
 
     def get_sim(self):
         return self._sandbox_service.sim
 
     def get_gui_controlled_agent_index(self):
         return self._gui_agent_ctrl._agent_idx
 
-    def _get_camera_lookat_pos(self):
-        agent_root = get_agent_art_obj_transform(
-            self.get_sim(), self.get_gui_controlled_agent_index()
-        )
-        lookat = agent_root.translation + mn.Vector3(0, 1, 0)
-        return lookat
-
-    def on_environment_reset(self, episode_recorder_dict):
-        base_pos = self._get_camera_lookat_pos()
-        self._camera_helper.update(base_pos, None)
+    def on_enter(self, prev_state, next_state):
+        self._camera_helper = next_state._camera_helper
```

Comment:
It seems like you're going to a lot of trouble to avoid writing some explicit transition-handling code in SandboxDriver like `self._app_state_tutorial = AppStateTutorial(..., self._app_state_rearrange.get_camera(), ...)` or `self._app_state_tutorial.set_target_camera(self._app_state_rearrange.get_camera())`.

I guess this is fine for now. But in general it doesn't seem safe to assume every AppState has a _camera_helper, and it's hacky to access a protected field in this way.

Put another way: an AppState class in general isn't going to have enough context to be able to manage a transition between App States.